Modules/_sqlite/connection.c

```diff
@@ -118,8 +118,13 @@ int pysqlite_connection_init(pysqlite_Connection* self, PyObject* args, PyObject
         return -1;
     }
     Py_BEGIN_ALLOW_THREADS
+#ifdef HAVE_SQLITE3_OPEN_V2
+    rc = sqlite3_open_v2(database, &self->db,
```

Comment:
I think that there is an opportunity to simplify this whole `SQLITE_OPEN_URI` and `HAVE_SQLITE3_OPEN_V2` dance.